src/Microsoft.AspNet.Server.Kestrel/Http/Connection.cs

```diff
@@ -29,15 +30,17 @@ public Connection(ListenerContext context, UvStreamHandle socket) : base(context
             ConnectionControl = this;
 
             _connectionId = Interlocked.Increment(ref _lastConnectionId);
+
+            SocketInput = new SocketInput(Memory2);
+            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
+            _frame = new Frame(this);
         }
 
         public void Start()
         {
             Log.ConnectionStart(_connectionId);
 
-            SocketInput = new SocketInput(Memory);
-            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
-            _frame = new Frame(this);
+            Task.Run((Action)_frame.ProcessFraming);
```

Comment:
Ok, measured a few permutations and stackwalked them; Task.Run does a bunch of stuff but ends up being ```ThreadPool.QueueUserWorkItem``` with extras; so cut to the chase here as you don't care about the extras.

Change ```ProcessFraming``` from ```async void``` to ```async Task``` as though it probably won't do any harm in this scenario, ```async void``` like the ```goto``` of the async world and people will make fun of you.

Then just jump straightish to ```QueueUserWorkItem``` as follows:

```csharp
public void Start()
{
    Log.ConnectionStart(_connectionId);

    ThreadPool.QueueUserWorkItem(Process, _frame.ProcessFraming);
    _socket.ReadStart(_allocCallback, _readCallback, this);
}

private void Process(object state)
{
    ((Func<Task>)state)();
}
```

Still question on ordering of ```_socket.ReadStart``` though?